CarCleaningService.rb

```diff
@@ -0,0 +1,145 @@
+class Car
+
+  attr_reader :start_time
+  attr_accessor :pickup_time, :car_time
+
+  def initialize()
+    @start_time = Time.now
+    #time when we start to wash your car
+    @car_time = nil
+    @pickup_time = nil
+  end
+end
+
+class Service 
+    OPEN_SERVICE = 8
+    CLOSE_SERVICE = 20
+    CAR_CLEANING_TIME = 2
+    def initialize()
+      @worker1 = Array.new
+      @worker2 = Array.new
+    end
+
+    def pickup(car)
+      #when both are free
+      if @worker1.empty?
+        car.car_time = car.start_time
+        pickup = car.car_time
+        pickup += CAR_CLEANING_TIME * 60 * 60
+        if pickup.hour < CLOSE_SERVICE || pickup.hour == CLOSE_SERVICE && pickup.min == 0
+          #today
+          car.pickup_time = pickup
+        else
+          #next day
+            pickup += 12 * 60 * 60
+            car.pickup_time = pickup
+          end
+        @worker1.push(car)
+        return
+      elsif @worker2.empty?
+        car.car_time = car.start_time
+        pickup = car.car_time
+        pickup += CAR_CLEANING_TIME * 60 * 60
+        if pickup.hour < CLOSE_SERVICE || pickup.hour == CLOSE_SERVICE && pickup.min == 0
+          #today
+          car.pickup_time = pickup
+        else
+          #next day
+            pickup += 12 * 60 * 60
+            car.pickup_time = pickup
+        end
+        @worker2.push(car)
+        return
```

Comment:
no need to explicitly use return here, also this method is very long and hard to follow, many if..else blocks. Can you maybe try to split the logic?